Sources/ShortcutFoundation/Core/Logger/Loggable.swift

```diff
@@ -0,0 +1,145 @@
+
+    /// Init Method that creates a custom logger for logging to a specific identifier and category.
+    /// - Author: Gabriel Sabadin
+    ///
+    /// - Parameter verbosity: The desired Verbosity
+    /// - Parameter identifier: A String to determine the identifier of the logger
+    /// - Parameter category: A String to determine the category of the logger
+    init(verbosity: Verbosity, identifier: String, category: String)
+
+    /// Method that triggers the logging procedure
+    /// - Author: Gabriel Sabadin
+    ///
+    /// - Parameter message: A String to determine the message you want to be logged
+    func log(message: String)
+}
+
+struct PrintLogger: Loggable {
+    var verbosity: Verbosity
+
+    init(verbosity: Verbosity, identifier: String, category: String) {
+        self.verbosity = verbosity
+    }
+
+    func log(message: String) {
+        switch verbosity {
+        case .debug:
+            print("🕵️‍♀️ DEBUG: \(message)")
+        case .info:
+            print("ℹ️ INFO: \(message)")
+        case .warning:
+            print("⚠️ WARNING: \(message)")
+        case .error:
+            print("⛔️ ERROR: \(message)")
+        case .critical:
+            print("☣️ CRITICAL: \(message)")
+        case .silent:
+            // no log message message
+            break
+        }
+    }
+}
+
+import os
+@available(iOS 14.0, *)
+struct AppleLogger: Loggable {
+    private let logger: os.Logger
+    var verbosity: Verbosity
+
+    init(verbosity: Verbosity, identifier: String, category: String) {
+        self.verbosity = verbosity
+        logger = os.Logger(subsystem: identifier, category: category)
+    }
+
+    func log(message: String) {
+        switch verbosity {
+        case .debug:
+            logger.debug("\(message)")
+        case .info:
+            logger.info("\(message)")
+        case .warning:
+            logger.warning("\(message)")
+        case .error:
+            logger.error("\(message)")
+        case .critical:
+            logger.critical("\(message)")
+        case .silent:
+            // no log message message
+            break
+        }
+    }
+}
+
+public struct Logger: Loggable {
+    private var strategy: Loggable
+
+    public var verbosity: Verbosity = .silent {
+        didSet {
+            strategy.verbosity = verbosity
+        }
+    }
+
+    public init(verbosity: Verbosity = .silent,
+                identifier: String = "",
+                category: String = "") {
+        if #available(iOS 14.0, *) {
+            strategy = AppleLogger(verbosity: verbosity, identifier: identifier, category: category)
+        } else {
+            strategy = PrintLogger(verbosity: verbosity, identifier: identifier, category: category)
+        }
+    }
+
+    private func makeStrategy(identifier: String = "",
+                              category: String = "") -> Loggable {
+        return PrintLogger(verbosity: verbosity, identifier: identifier, category: category)
+    }
```

Comment:
Where can we use this method?

Reply:
@darya-gurinovich nowhere, I was considering removing the IF statement to this method, but changed my mind and forgot to delete it. Thanks for noticing.